### ISSUE TITLE
支持训练时在多个验证集上独立评估

### ISSUE THREAD

**Issue by PolarisHsu** [modelscope/ms-swift#5467]
**Describe the feature**
现在的val_dataset参数支持传入多个验证集，但会将它们concat一起评估，无法直观看到在不同的验证集上的表现。希望能支持独立评估，在每个验证集上分别报告metrics。

**Paste any useful information**
transformers Trainer的eval_dataset参数本身支持传入一个字典：dict[str, torch.utils.data.Dataset]，详见[官方文档](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Trainer.eval_dataset)

**Comment by Jintao-Huang** [comment]
可以使用channel loss

**Comment by PolarisHsu** [comment]
channel loss似乎是为训练集而准备？对于验证集感觉不是很合适

**Comment by WallE-Chang** [comment]
+1

**Comment by Jintao-Huang** [comment]
channel loss 不行吗

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to create a new issue if it the problem persists.
